tccgen.c

```diff
@@ -319,11 +319,6 @@ static int RC2_TYPE(int t, int rc)
 #endif
     if (rc & RC_FLOAT)
         return RC_FLOAT;
-// advance to the next register class for two argument calls
-#if defined TCC_TARGET_RISCV32 || defined TCC_TARGET_RISCV32
-    if (rc > RC_FLOAT)
-        return rc << 1;
-#endif
```

Comment:
This code is why manual mv instructions are required to fix up two register argument passing. A better check is required so that if a non-existent register class is attempted, it gets pushed to the stack. That would eliminate the fix up code as well.